spec/lib/efile/nj/nj1040_calculator_spec.rb

```diff
@@ -1055,6 +1086,23 @@ def over_65_birth_year
     end
   end
 
+  describe 'line 54 - total tax due' do
+    let(:intake) { create(:state_file_nj_intake) }
+    it 'sets line 54 to equal line 50 plus line 51' do
+      allow(instance).to receive(:calculate_line_50).and_return 20_000
+      allow(instance).to receive(:calculate_line_51).and_return 8_000
+      instance.calculate
+      expect(instance.lines[:NJ1040_LINE_54].value).to eq(28_000)
+    end
+
+    it 'sets line 54 to 0 if the sum is negative' do
+      allow(instance).to receive(:calculate_line_50).and_return -20_000
+      allow(instance).to receive(:calculate_line_51).and_return 10_000
+      instance.calculate
+      expect(instance.lines[:NJ1040_LINE_54].value).to eq(0)
+    end
```

Comment:
[dust] is this realistic since line 50 (and i think all the ones that could eventually go into 54) also is explicitly never lower than 0? there's not really a big downside to having an extra test case other than overall code reduction, so up to you

Reply:
Hm, I didn't think about it. I just put it in because the PMs had put it explicitly in the acceptance criteria.

Reply:
A great point, I didn't think about it... It was just explicitly stated in the story description ("if less than zero, it should be zero") so I just added a test/behavior for that without thinking

Reply:
Discussion resolution - we'll keep this behavior for now to future-proof the more extensive scope for this line. [Slack thread](https://njcio.slack.com/archives/C0590CD1EE8/p1732304688191179)